### ISSUE TITLE
Expose path parameters as context variables in HTTP security matchers

### ISSUE THREAD

**Issue by chrylis** [spring-projects/spring-security#4084]
I am trying to configure a security expression for path `/api/users/{id}/foo` such that only user `id` can read it; that is, `principal.id == #id`. I can do this with an `@PreAuthorize` on a controller method, but I'm attempting to shoehorn this onto a gap in Spring Data REST by writing it in the filter configuration:

```
http
    .authorizeRequests()
        .mvcMatchers("/api/users/{id}/foo").access("hasRole('ADMIN') or principal.id == #id")
        .antMatchers("/api/**").authenticated();
```

I had hoped that in this case the `{id}` from the URI template would be exposed to the access expression, but it isn't. As an enhancement, it would be helpful to expose the matching template parameters. I'm aware that matching against the conversion service is not feasible, but even exposing them as a string would be sufficient here.

**Comment by rwinch** [comment]
What version of Spring Security are you using? Spring Security 4.1.0.RELEASE+ added support for path variables. See http://docs.spring.io/spring-security/site/docs/4.1.x/reference/htmlsingle/#el-access-web-path-variables

**Comment by rwinch** [comment]
Closing due to no additional feedback and that this is resolved in Spring Security 4.1+
